### PR TITLE
Bugfix custom cache

### PR DESCRIPTION
**Pull request by a-szotyori**
When a custom cache was set the RequestQueue was not started.

Using the RequestQueueBuilder again to create RequestQueues and moved certificate handling to this builder.

Updated Volley to 1.1.0.

Extended the build to run tests on an emulator.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,9 +3,13 @@
...
     repositories {
         jcenter()
+        maven {
+            url 'https://maven.google.com/'
+            name 'Google'
+        }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.0'
+        classpath 'com.android.tools.build:gradle:2.3.3'
```
Comment (EarlOfEgo) [inline build.gradle]: what's the reason for not updating to `3.0.1`?

File: build.gradle [hunk 2]
```diff
@@ -3,9 +3,13 @@
 buildscript {
     repositories {
         jcenter()
+        maven {
```
Comment (EarlOfEgo) [inline build.gradle]: with gradle 3.0.1 you could use `google()` instead

File: build.gradle (hunk 1 above)
Comment (a-szotyori) [reply build.gradle]: Good point! We'll update next time! 👍

Comment (EarlOfEgo) [reply build.gradle]: [here](https://github.com/gini/gini-sdk-android/pull/5) we go :)

**Pull request merged**